editor/plugins/script_editor_plugin.cpp

```diff
@@ -1380,33 +1380,57 @@ void ScriptEditor::_update_script_names() {
 	_update_script_colors();
 }
 
-void ScriptEditor::edit(const Ref<Script> &p_script, bool p_grab_focus) {
+bool ScriptEditor::edit(const Ref<Script> &p_script, int p_line, int p_col, bool p_grab_focus) {
 
 	if (p_script.is_null())
-		return;
+		return false;
 
 	// refuse to open built-in if scene is not loaded
 
 	// see if already has it
 
 	bool open_dominant = EditorSettings::get_singleton()->get("text_editor/files/open_dominant_script_on_scene_change");
 
+	Error err = p_script->get_language()->open_in_external_editor(p_script, p_line >= 0 ? p_line : 0, p_col);
+	if (err == OK)
+		return false;
+	if (err != ERR_UNAVAILABLE)
+		WARN_PRINT("Couldn't open in custom external text editor");
+
 	if (p_script->get_path().is_resource_file() && bool(EditorSettings::get_singleton()->get("text_editor/external/use_external_editor"))) {
 
 		String path = EditorSettings::get_singleton()->get("text_editor/external/exec_path");
 		String flags = EditorSettings::get_singleton()->get("text_editor/external/exec_flags");
+
+		Dictionary keys;
+		keys["project"] = GlobalConfig::get_singleton()->get_resource_path();
+		keys["file"] = GlobalConfig::get_singleton()->globalize_path(p_script->get_path());
+		keys["line"] = p_line >= 0 ? p_line : 0;
```

Comment:
Here it's needed though.
I think this is kind of a convention among some text editors. For example, in VSCode `file:0:0` is the same as `file`. If the file is already open it won't change the line. Same goes for MonoDevelop. For Atom, on the other hand, it's equivalent to `file:1:0` so it takes you to the first line, which kind of sucks (you don't want it to change the line unless you explicitly requested that, like with goto_method).
I made it like this to avoid having two separated settings, `exec_flags` which does not accept line and col (e.g.: `{project} {file}), and `exec_at_line_flags` which is the same as above.